src/BlockTypes/OrderConfirmation/AbstractOrderConfirmationBlock.php

```diff
@@ -0,0 +1,207 @@
+
+	/**
+	 * This is what gets rendered when the order does not exist. Renders nothing by default, but can be overridden by
+	 * child classes.
+	 *
+	 * @return string
+	 */
+	protected function render_content_fallback() {
+		return '';
+	}
+
+	/**
+	 * Get current order.
+	 *
+	 * @todo Check auth.
+	 *
+	 * @return \WC_Order|null
+	 */
+	protected function get_order() {
+		$order_id = absint( get_query_var( 'order-received' ) );
+
+		if ( $order_id ) {
+			return wc_get_order( $order_id );
+		}
+
+		return null;
+	}
+
+	/**
+	 * View mode for order details based on the order, current user, and settings.
+	 *
+	 * Possible values are:
+	 * - "full" user can view all order details.
+	 * - "limited" user can view some order details, but no PII. This may happen for example, if the user checked out as a guest.
+	 * - false user cannot view order details.
+	 *
+	 * @param \WC_Order|null $order Order object.
+	 * @return "full"|"limited"|false
+	 */
+	protected function get_view_order_permissions( $order ) {
+		if ( ! $order ) {
+			return false;
+		}
+
+		if ( is_user_logged_in() ) {
+			// If logged in, check the user owns the order.
+			return $this->is_current_customer_order( $order ) ? 'full' : false;
+		}
+
+		// If the user is logged out, check the order key for validity.
+		if ( $this->allow_guest_checkout() && $this->is_current_customer_order( $order ) && $this->has_valid_order_key( $order ) ) {
+			return $this->order_matches_session( $order ) ? 'full' : 'limited';
+		}
+
+		return false;
+	}
+
+	/**
+	 * See if guest checkout is enabled.
+	 *
+	 * @return boolean
+	 */
+	protected function allow_guest_checkout() {
+		return 'yes' === get_option( 'woocommerce_enable_guest_checkout' );
+	}
+
+	/**
+	 * See if the order key is valid.
+	 *
+	 * @param \WC_Order $order Order object.
+	 * @return boolean
+	 */
+	protected function has_valid_order_key( $order ) {
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		return ! empty( $_GET['key'] ) && $order->key_is_valid( wc_clean( wp_unslash( $_GET['key'] ) ) );
+	}
+
+	/**
+	 * See if the current user ID matches the given order customer ID.
+	 *
+	 * @param \WC_Order $order Order object.
+	 * @return boolean
+	 */
+	protected function is_current_customer_order( $order ) {
+		return $order->get_user_id() === get_current_user_id();
+	}
+
+	/**
+	 * See if the order matches the current session.
+	 *
+	 * @param \WC_Order $order Order object.
+	 * @return boolean
+	 */
+	protected function order_matches_session( $order ) {
+		return $order->get_id() === wc()->session->get( 'store_api_draft_order' );
+	}
+
+	/**
+	 * Get the frontend script handle for this block type.
+	 *
+	 * @param string $key Data to get, or default to everything.
+	 */
+	protected function get_block_type_script( $key = null ) {
+		return null;
+	}
+
+	/**
+	 * Get a fake order for previews.
+	 *
+	 * @return \WC_Order Fake order.
```

Comment:
There's a fullstop behind the return type. This is only required for comments, not for return values.

```suggestion
	 * @return \WC_Order Fake order
```